### PR TITLE
wp-surface-scale-v1: Initial implementation

### PR DESCRIPTION
**Pull request by kennylevinsen**
Depends on https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/3511 and https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/143

### CODE REVIEW

File: protocols/meson.build [hunk 1]
```diff
@@ -20,6 +20,7 @@ protocols = [
 	['idle.xml'],
 	['wlr-input-inhibitor-unstable-v1.xml'],
 	['wlr-output-power-management-unstable-v1.xml'],
+	['wp-surface-scale-v1.xml'],
```
Comment (emersion) [inline protocols/meson.build:23]: We shouldn't need the protocol here if generated enums aren't used in wlroots' public header.

Comment (kennylevinsen) [reply protocols/meson.build]: Ah, good point.

File: sway/tree/container.c [hunk 2]
```diff
@@ -1325,6 +1326,7 @@ static void surface_send_enter_iterator(struct wlr_surface *surface,
 		int x, int y, void *data) {
 	struct wlr_output *wlr_output = data;
 	wlr_surface_send_enter(surface, wlr_output);
+	wlr_fractional_scale_v1_notify_scale(surface, wlr_output->scale);
```
Comment (emersion) [inline sway/tree/container.c:1329]: Hm, should we iterate over all outputs and send the highest scale here?

**Comment by emersion** [comment]
There are a few other places where we call `wlr_surface_send_enter()`, but don't update the fractional scale, e.g. layer-shell.

**Comment by emersion** [comment]
This PR has some useful bits which could be leveraged here: https://github.com/swaywm/sway/pull/7443

**Comment by emersion** [comment]
I've created an alternative PR here: #7448

**Pull request closed without merging**